components/logins/ios/Logins/LoginsStorage.swift

```diff
@@ -94,26 +135,53 @@ open class LoginsStorage {
         }
     }
 
+    /// Unlock the database.
+    /// `key` can be any string as it is a passphrase (e.g. `bobo`).
+    /// `salt` must be an hex-encoded string of 32 characters (e.g. `a6a97a03ac3e5a20617175355ea2da5c`).
+    ///
+    /// Throws `LockError.mismatched` if the database is already unlocked.
+    ///
+    /// Throws a `LoginStoreError.InvalidKey` if the key is incorrect, or if dbPath does not point
+    /// to a database, (may also throw `LoginStoreError.Unspecified` or `.Panic`).
+    open func unlockWithKeyAndSalt(key: String, salt: String) throws {
+        try queue.sync {
+            if self.raw != 0 {
+                throw LockError.mismatched
+            }
+            try self.doOpen(key, salt: salt)
+        }
+    }
+
+    /// Equivalent to `unlockWithKeyAndSalt(key:, salt:)`, but does not throw if the
+    /// database is already unlocked.
+    open func ensureUnlockedWithKeyAndSalt(key: String, salt: String) throws {
+        try queue.sync {
+            try self.doOpen(key, salt: salt)
+        }
+    }
+
     /// Unlock the database.
     ///
     /// Throws `LockError.mismatched` if the database is already unlocked.
     ///
     /// Throws a `LoginStoreError.InvalidKey` if the key is incorrect, or if dbPath does not point
     /// to a database, (may also throw `LoginStoreError.Unspecified` or `.Panic`).
+    @available(*, deprecated, message: "Use unlockWithKeyAndSalt instead.")
```

Comment:
Hrm, I'm not sure if we should outright deprecate this, or if we should just leave both options available. Are there cases where swift consumers might not want to use the plaintext-header approach?

Reply:
I think it's fair to assume that this code will be used only on iOS for the medium term. We can always revisit this decision anyway and as you said it steers the current iOS consumers of the framework in the "right" direction.